akd/src/proto/mod.rs

```diff
@@ -60,6 +60,17 @@ macro_rules! require {
     };
 }
 
+macro_rules! require_messagefield {
+    ($obj:ident, $field:ident) => {
+        if !$obj.$field.is_some() {
+            return Err(LocalAuditorError::RequiredFieldMissing(
+                stringify!($obj).to_string(),
+                stringify!($has_field).to_string(),
+            ));
+        }
+    };
+}
+
```

Comment:
It's odd that struct fields need to be handled differently from native type fields, but from a quick look at the library docs I couldn't find a unified API, so this is probably the best we can do for now.

Reply:
yeah I spent a good amount of time on this, and I don't see a workaround unfortunately :(. It'll be fun with the other protobufs we have...